tracer/src/Datadog.Trace/TraceContext.cs

```diff
@@ -92,30 +96,24 @@ internal void EnableIastInRequest()
         {
             if (_iastRequestContext is null)
             {
-                lock (_syncRoot)
-                {
-                    _iastRequestContext ??= new();
-                }
+                Interlocked.CompareExchange(ref _iastRequestContext, new(), null);
```

Comment:
Just a note that this potentially creates multiple `IastRequestContext`s, if two threads enter the `if` at the same time. Only one is set on the `TraceContext` obviously, but just want to check that creating two `IastRequestContext` instances (and throwing one away immediately) won't be an issue?

Reply:
True, I think it shouldn't be so much of an issue here, because this will rarely happen. This method is called after the 
`AcquireRequest` lock. And if happens I prefer to trash a new instance and keep the previous `_iastRequestContext` instance.

<img width="1156" alt="image" src="https://user-images.githubusercontent.com/69803/237060396-0be4645c-088d-41bf-9b8e-ac8c54ef486b.png">

To reduce the odds I will add a volatile read in the `if`

Reply:
done in [84ce8de](https://github.com/DataDog/dd-trace-dotnet/pull/4125/commits/84ce8de1223471d73f08d97452e7fdf9cd483075)

Reply:
Yeah, my initial concern was whether `IastRequestContext` had any static state, so creating two would be an issue, but I should have just checked the code - it doesn't 😄 So I agree, this isn't a big concern